packages/ai/src/requests/stream-reader.ts

```diff
@@ -197,15 +202,19 @@ export function aggregateResponses(
          * Candidates should always have content and parts, but this handles
          * possible malformed responses.
          */
-        if (candidate.content && candidate.content.parts) {
+        if (candidate.content) {
+          // Skip a candidate without parts.
+          if (!candidate.content.parts) {
+            continue;
+          }
           if (!aggregatedResponse.candidates[i].content) {
             aggregatedResponse.candidates[i].content = {
               role: candidate.content.role || 'user',
               parts: []
             };
           }
-          const newPart: Partial<Part> = {};
```

Comment:
This is the part where I'm not sure why we did it like that instead of the way I have changed it to.